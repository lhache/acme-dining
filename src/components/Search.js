import React, { Component } from 'react';
import './Search.css';

class Search extends Component {

  constructor(props) {
    super(props)
    this.state = {
      text: ''
    }

    this._handleTextChange = this._handleTextChange.bind(this)
    this._handleSubmit = this._handleSubmit.bind(this)
  }

  _handleTextChange(event) {
    this.setState({text: event.target.value})
  }

  _handleSubmit(event) {
    event.preventDefault()
    this.props.search(this.state.text)
  }

  render() {
    return (
      <div className="Search">
        <form onSubmit={this._handleSubmit}>
          <input type="text" onChange={this._handleTextChange} />
          <button type="submit" ref="button">Search</button>
        </form>
      </div>
    );
  }
}

export default Search;
