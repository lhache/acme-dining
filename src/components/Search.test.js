import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { withApollo } from 'react-apollo';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Search from './Search';

const search = () => {
  return true;
}

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const wrapper = shallow(<Search search={search} />)
  expect(wrapper).toMatchSnapshot();
});

it('has the default text state as empty string', () => {
  const wrapper = shallow(<Search search={search} />)
  expect(wrapper.state().text).toBe('');
});

it('has the props search of type function', () => {
  const wrapper = shallow(<Search search={search} />)
  const instance = wrapper.instance()
  expect(instance.props.search).toBeTruthy()

});

it('has the text state changing when there is a change on the input', () => {
  const wrapper = shallow(<Search search={search} />)


});

it('calls _handleSubmit on form submit', () => {

  const wrapper = shallow(<Search search={search} />)
  const _handleSubmit = jest.fn()
  wrapper._handleSubmit = _handleSubmit

  wrapper.find('button').simulate('click');
  expect(wrapper._handleSubmit).toHaveBeenCalled()


});
