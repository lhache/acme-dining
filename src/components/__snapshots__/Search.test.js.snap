// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    search={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <div>
          <input
            onChange={[Function]}
            type="text"
          />
          <button
            type="submit"
          >
             Search
          </button>
        </div>
      </form>,
      "className": "Search",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <input
            onChange={[Function]}
            type="text"
          />
          <button
            type="submit"
          >
             Search
          </button>
        </div>,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              type="text"
            />,
            <button
              type="submit"
            >
               Search
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": " Search",
              "type": "submit",
            },
            "ref": "searchButton",
            "rendered": " Search",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <div>
            <input
              onChange={[Function]}
              type="text"
            />
            <button
              type="submit"
            >
               Search
            </button>
          </div>
        </form>,
        "className": "Search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <input
              onChange={[Function]}
              type="text"
            />
            <button
              type="submit"
            >
               Search
            </button>
          </div>,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                type="text"
              />,
              <button
                type="submit"
              >
                 Search
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": " Search",
                "type": "submit",
              },
              "ref": "searchButton",
              "rendered": " Search",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
